In [1]:
#Q1. What is an ensemble technique in machine learning?

Ensemble technique in machine learning is a method of combining multiple models to improve the overall performance of the model. It is defined as a multimodal system in which different classifiers and techniques are strategically combined into a predictive model

In [2]:
#Q2. Why are ensemble techniques used in machine learning?

Ensemble methods help to reduce variance in predicted data, minimize biasness in predictive models, and classify and predict statistics from complex problems with better accuracy. It also reduces the risk of overfitting

In [ ]:
#Q3. What is bagging?

Bagging, also known as Bootstrap aggregating, is an ensemble learning technique that helps to improve the performance and accuracy of machine learning algorithms

Bagging works by creating multiple subsets of data from the original dataset using bootstrapping. 

Each subset of data is then used to train a separate model. The final prediction is made by averaging the predictions of all models

In [3]:
#Q4. What is boosting?

It is a family of machine learning algorithms that convert weak learners to strong ones. Boosting works by iteratively training a sequence of weak models on repeatedly modified versions of the data 

Boosting helps to increase the accuracy of the model. It is used with decision trees but can also be applied to other algorithms 

In [4]:
#Q5. What are the benefits of using ensemble techniques?

Ensemble methods are used because they:


Increase performance

Reduce risk of overfitting

Improve robustness

Boost overall accuracy

In [5]:
#Q6. Are ensemble techniques always better than individual models?

In summary, ensemble techniques are not always better than individual models but they generally perform better than single models when the individual classifier is weak

In [ ]:
#Q7. How is the confidence interval calculated using bootstrap?

The bootstrap method is a resampling technique that involves repeatedly drawing samples from a dataset and estimating a statistic of interest for each sample. The confidence interval is then calculated based on the distribution of these estimates

In [6]:
#Q8. How does bootstrap work and What are the steps involved in bootstrap?

Bootstrapping is a method of inferring results for a population from smaller random samples with replacement

Steps involved are below:

Step 1: Draw a sample of size N from the original dataset with replacement. This is a bootstrap sample.

Step 2:Repeat step 1 N times, so that we have N bootstrap samples.

Step 3: Estimate our value on each of the bootstrap samples, so that we have N estimates

Step 4: Use the distribution of estimates for inference (for example, estimating the confidence intervals).

In [7]:
#Q9. A researcher wants to estimate the mean height of a population of trees. They measure the height of a
#sample of 50 trees and obtain a mean height of 15 meters and a standard deviation of 2 meters. Use
#bootstrap to estimate the 95% confidence interval for the population mean height.

In [10]:
import numpy as np


# Sample data
data_sample = np.random.normal(loc=15, scale=2, size=1000)

#define number of samples

n_boot= 5000

boot_straps= np.random.choice(data_sample, size=(n_boot,len(data_sample)),replace=True)
bootstrap_means = boot_straps.mean(axis=1)
bootstrap_std = boot_straps.std()

# Calculate confidence interval
conf_int = (bootstrap_means.mean() - 1.96 * bootstrap_std, bootstrap_means.mean() + 1.96 * bootstrap_std)

print("95% Confidence Interval:", conf_int)



95% Confidence Interval: (11.124417906907816, 18.828636507902786)


We expect that 95% of the times the height of trees will fall between (11.12 and 18.83)